release.py

```diff
@@ -479,16 +479,16 @@ def select_gpg_key():
 for root, dirs, files in os.walk(artifacts_dir):
     assert root.startswith(artifacts_dir)
 
-    for file in files:
-        local_path = os.path.join(root, file)
-        remote_path = os.path.join("public_html", kafka_output_dir, root[len(artifacts_dir)+1:], file)
-        sftp_cmds += "\nput %s %s" % (local_path, remote_path)
-
     for dir in dirs:
         sftp_mkdir(os.path.join("public_html", kafka_output_dir, root[len(artifacts_dir)+1:], dir))
 
-if sftp_cmds:
-    cmd("Uploading artifacts in %s to your Apache home directory" % root, "sftp -b - %s@home.apache.org" % apache_id, stdin=sftp_cmds)
+    for file in files:
+        local_path = os.path.join(root, file)
+        remote_path = os.path.join("public_html", kafka_output_dir, root[len(artifacts_dir)+1:], file)
+        sftp_cmds = """
+put %s %s
+""" % (local_path, remote_path)
+        cmd("Uploading artifacts in %s to your Apache home directory" % root, "sftp -b - %s@home.apache.org" % apache_id, stdin=sftp_cmds)
```

Comment:
Is there a reason we needed to rearrange this? I think it was already slow and when I had it doing individual commands for each action it took even longer.

Reply:
Yeah it is very slow and `put -r` should be much faster. I recall there is comment that says we can not do `put -r` due to bug in some version. `put -r` in sftp works for me on my RHEL 7 desktop. I might make sense to provide a commented code path in the release.py so that we can optionally choose to use `put -r` if it is supported.

I recall Matthias said in the other PR that this change is to make the sftp upload more interactive.

Reply:
makes sense, probably ideal solution would be to just get the output from commands printing incrementally instead of all at once at the end (or something more efficient than the sftp we are currently stuck with...).

Reply:
> probably ideal solution would be to just get the output from commands printing incrementally instead of all at once at the end

Cannot follow here. With the change, we call `sftp` per file and report progress per file.

Reply:
it's fine as is, I was just saying that if we could stream a single sftp processes output, that would be faster & more efficient.

Reply:
https://github.com/apache/kafka/pull/9070 apparently reduces the time from 2 hours to 5 minutes. sftp per file seems pretty slow even for something that is not used frequently. Not sure if it became slower over time or something specific to John's connection though. :)

Reply:
It was always slow. This PR only made the script print stuff to stdout regularly--before this PR, the script was uploading silently for a loooong time and one could think it froze (even if it was just doing/uploading stuff). This PR was not to make the upload faster, just to make the script appear alive :)